langlib/lang.__internal/src/main/java/org/ballerinalang/langlib/internal/GetAttribute.java

```diff
@@ -54,6 +54,10 @@ public static Object getAttribute(Strand strand, XMLValue xmlVal, String attrNam
                     "Invalid xml attribute access on xml " + xmlVal.getNodeType().value());
         }
         XMLQName qname = new XMLQName(attrName);
-        return xmlVal.getAttribute(qname.getLocalName(), qname.getUri());
+        String attrVal = xmlVal.getAttribute(qname.getLocalName(), qname.getUri());
+        if (attrVal == null) {
+            return createError(XML_OPERATION_ERROR, "attribute '" + attrName + "' not found");
```

Comment:
As discussed offline, we would have to consider optional access here right?

Shall we also add some tests?

Reply:
+1